### PR TITLE
Release v4.0.0

### PR DESCRIPTION
**Pull request by mr-tz**
```mermaid
graph TD;
v3-->v4
```

# Release checklist

- [x] Ensure all [milestoned issues/PRs](https://github.com/mandiant/capa/milestones) are addressed, or reassign to a new milestone.
- [x] Add the `dont merge` label to all PRs that are close to be ready to merge (or merge them if they are ready) in [capa](https://github.com/mandiant/capa/pulls) and [capa-rules](https://github.com/mandiant/capa-rules/pulls).
- [x] Ensure the [CI workflow succeeds in master](https://github.com/mandiant/capa/actions/workflows/tests.yml?query=branch%3Amaster).
- [x] Ensure that `python scripts/lint.py rules/ --thorough` succeeds (only `missing examples`  offenses are allowed in the nursery).
- [x] Review changes
  - capa https://github.com/mandiant/capa/compare/v3.2.0...master
  - capa-rules https://github.com/mandiant/capa-rules/compare/v3.2.0...master
- [x] Update [CHANGELOG.md](https://github.com/mandiant/capa/blob/master/CHANGELOG.md)
  - Do not forget to add a nice introduction thanking contributors
  - Remember that we need a major release if we introduce breaking changes
  - Sections: see template below
  - Update `Raw diffs` links
  - Create placeholder for `master (unreleased)` section
- [x] Update [capa/version.py](https://github.com/mandiant/capa/blob/master/capa/version.py)
- [x] Create a PR with the updated [CHANGELOG.md](https://github.com/mandiant/capa/blob/master/CHANGELOG.md) and [capa/version.py](https://github.com/mandiant/capa/blob/master/capa/version.py). Copy this checklist in the PR description.
- [ ] After PR review, merge the PR and [create the release in GH](https://github.com/mandiant/capa/releases/new) using text from the [CHANGELOG.md](https://github.com/mandiant/capa/blob/master/CHANGELOG.md).
- [ ] Verify GH actions [upload artifacts](https://github.com/mandiant/capa/releases), [publish to PyPI](https://pypi.org/project/flare-capa) and [create a tag in capa rules](https://github.com/mandiant/capa-rules/tags) upon completion.
- [ ] Manually update capa rules major version rule branch
  ```commandline
  [capa/rules] $ git pull master
  [capa/rules] $ git checkout v3  # create if new major version: git checkout -b vX
  [capa/rules] $ git merge master
  [capa/rules] $ git push origin v3
  ```
- [ ] [Spread the word](https://twitter.com)
- [ ] Update internal service

### CODE REVIEW

**Comment by mike-hunhoff** [comment]
See branch conflicts - unsure what's happening there.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,37 @@
...
+### capa explorer IDA Pro plugin
+
+### Development
+
+### Raw diffs
+- [capa v4.0.0...master](https://github.com/mandiant/capa/compare/<release>...master)
+- [capa-rules v4.0.0...master](https://github.com/mandiant/capa-rules/compare/<release>...master)
+
+## v4.0.0 (2022-07-XX)
+Version 4 adds support for analyzing .NET executables. For this a new feature extractor and an updated rule syntax now support .NET features including `namespace`, `class`, `api`, and `string`.
```
Comment (williballenthin) [inline CHANGELOG.md:24]:
```suggestion
Version 4 adds support for analyzing .NET executables. capa will autodetect .NET modules, or you can explicitly invoke the new feature extractor via `--format dotnet`. We've also extended the rule syntax for .NET features including `namespace` and `class`.
```

File: capa/main.py [hunk 2]
```diff
@@ -518,6 +518,7 @@ def get_extractor(
 
         import capa.features.extractors.smda.extractor
 
+        logger.warning("Deprecation warning: v4.0 will be the last capa version to support the SMDA backend.")
```
Comment (williballenthin) [inline capa/main.py:521]: this logging strategy is good. we may also want to consider using the `warnings` module (https://docs.python.org/3/library/warnings.html) with a `DeprecationWarning` instance.

https://docs.python.org/3/library/exceptions.html#DeprecationWarning

**Review by williballenthin — APPROVED** [review]

File: capa/main.py (hunk 2 above)
Comment (mr-tz) [reply capa/main.py:522]: added in f4d0e8d

File: capa/features/address.py [hunk 3]
```diff
@@ -71,6 +71,10 @@ def __hash__(self):
     def __repr__(self):
         return f"token(0x{self.token.value:x})"
 
+    def __index__(self):
+        # returns the object converted to an integer
+        return self.token.value
```
Comment (mr-tz) [inline capa/features/address.py:76]: with these we can continue to use `"0x%x" % f.address`

Comment (mr-tz) [reply capa/features/address.py:76]: alternatively need to update usages and maybe use a format helper

**Pull request merged**